src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -141,14 +141,18 @@ class ReferralService(
     overrideCreatedAt: OffsetDateTime? = null,
     endOfServiceReport: EndOfServiceReport? = null,
   ): Referral {
+    val intervention = interventionRepository.getOne(interventionId)
+    val serviceCategories = intervention.dynamicFrameworkContract.contractType.serviceCategories
+    val selectedServiceCategories = if (serviceCategories.size == 1) serviceCategories.toMutableList() else mutableListOf()
```

Comment:
I thought it was best to assign the single service category on creating the referral if it's a non-cohort referral as it would otherwise require a separate call from the front-end to populate it, additional it's considered an invalid referral if it has no selected service category for non-cohort referral

Reply:
yes absolutely. why is this cast to a `mutableList` vs a plain `list`?

Reply:
toList() does not include add() method which is required by hibernate otherwise it throws an `unsupportedOperationException`

Reply:
interesting - how come we dont' have to set the data type to MutableList on the entity class?

Reply:
The datatype `List` is extended by `MutableList`. However the listOf() method returns a `EmptyList` object which doesn't contain `add()` methods